### PR TITLE
Make it easier for RESTEasy Reactive users to handle Hibernate related exceptions

### PR DESCRIPTION
**Pull request by geoand**

### CODE REVIEW

File: extensions/hibernate-reactive/deployment/src/main/java/io/quarkus/hibernate/reactive/deployment/ResteasyReactiveServerIntegrationProcessor.java [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+import javax.persistence.PersistenceException;
+
+import io.quarkus.deployment.annotations.BuildStep;
+import io.quarkus.resteasy.reactive.server.spi.UnwrappedExceptionBuildItem;
+
+public class ResteasyReactiveServerIntegrationProcessor {
+
+    @BuildStep
+    public UnwrappedExceptionBuildItem unwrappedExceptions() {
+        return new UnwrappedExceptionBuildItem(PersistenceException.class);
```
Comment (geoand) [inline extensions/hibernate-reactive/deployment/src/main/java/io/quarkus/hibernate/reactive/deployment/ResteasyReactiveServerIntegrationProcessor.java:12]: I am wondering if this should also include `HibernateException`

File: extensions/hibernate-orm/deployment/src/main/java/io/quarkus/hibernate/orm/deployment/ResteasyReactiveServerIntegrationProcessor.java [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+import javax.persistence.PersistenceException;
+
+import io.quarkus.deployment.annotations.BuildStep;
+import io.quarkus.resteasy.reactive.server.spi.UnwrappedExceptionBuildItem;
+
+public class ResteasyReactiveServerIntegrationProcessor {
+
+    @BuildStep
+    public UnwrappedExceptionBuildItem unwrappedExceptions() {
+        return new UnwrappedExceptionBuildItem(PersistenceException.class);
```
Comment (geoand) [inline extensions/hibernate-orm/deployment/src/main/java/io/quarkus/hibernate/orm/deployment/ResteasyReactiveServerIntegrationProcessor.java:12]: I am wondering if this should also include `HibernateException`

**Comment by Sanne** [comment]
Sorry I'm totally out of context :) Could you summarize what this is for?

**Comment by geoand** [comment]
Or course!

The idea is to make it very easy for a user to "handle" (in the JAX-RS sense) the actual root cause of an exception.

Take for example the Hibernate ORM quickstart, where trying to save a new Fruit with an existing name leads to a ConstraintViolationException. If a user currently wants to respond to HTTP 400 for that exception, they need handle the RollbackException, then look of the cause is a PersistenceException and check if that one's cause is a ConstraintViolationException.
With this change, writing the JAX-RS ExceptionMapper for that exception is straight forward for the user.

Does that make sense?

**Review by stuartwdouglas — APPROVED** [review]

**Comment by geoand** [comment]
@Sanne is this good for you as well?

**Review by Sanne — APPROVED** [review]
Sure, thanks

**Comment by geoand** [comment]
👍🏼
